org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AcquisitionDevice.java

```diff
@@ -561,13 +567,33 @@ public void pause() throws ScanningException {
 	}
 	
 	@Override
-	public void seek(int stepNumber) throws ScanningException {
-		// TODO FIXME The positioner should seek back to the previous position!!
+	public void seek(int stepNumber) throws ScanningException, InterruptedException {
+		// This is the values of all motors at this global (including malcolm) scan
+		// position. Therefore we do not need a subscan moderator but can run the iterator
+		// to the point
+		IPosition pos = positionForStep(stepNumber);
+		if (pos == null) throw new ScanningException("Seek position is invalid "+stepNumber);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)